### PR TITLE
125 field updates

### PR DESCRIPTION
**Pull request by ethanjohnsrud**
This one grew, as I found more issues.  May be easier to view per individual commit.

# #125 & #126 | Portal Sync Fixes
1. Sync Config Files
2. Add Checkbox for User Demo Profile
    1. Corrected Form Handling for header/footer children & Keys
3. Improve `InitialAccountFlow` handling based on role
4. Models Syncing
    1. Refactor Model Route Syncing, on initial load (All are on the same method now)
    2. Display Blank Loading Page; instead of blank fields flashing
    3. Implemented `ModelPopUpAction` enum to track state popup windows
6. Created `FullImagePage` 
    1. Renders blank for brief second between page load and JWT authentication
    4. Replaces `NotFound` page (404 tomb)
7. Animation plays on page load from link
    1. Also handles redirect query parameter
8. Redux: Local Storage Persistence for JWT and Settings
9. Added Loading Page to Models.  Displays for a second while fields are fetched from the server.

Screenshot 2024-10-13 153028

### CODE REVIEW

File: src/100-App/redux-store.tsx [hunk 1]
```diff
@@ -137,40 +153,85 @@ export const logoutAccount = async(dispatch: (arg0: {type: 'account/resetAccount
...
+} = settingsSlice.actions;
+
 
+export const initializeSettingsState = async(dispatch: (arg0: { payload: SettingsState; type: 'settings/setSettings'; }|{type: 'settings/resetSettings'; }) => void, getState: AccountState) => {
+    try {
+        const localStorageSettings:string|null = localStorage.getItem('settings');
+        const savedSettings:SettingsState = localStorageSettings ? JSON.parse(localStorageSettings) : initialSettingsState;
+        if(!isNaN(savedSettings.version) && (savedSettings.version == parseInt(process.env.REACT_APP_SETTINGS_VERSION ?? '1', 10)))
+          dispatch(setSettings({ ...initialSettingsState, ...savedSettings }));
+
```
Comment (bohnerjosh) [inline src/100-App/redux-store.tsx]: Probably would be best to be consistent here between mobile and the portal. We should only include initialSettingsState in setState if the state version in saved settings and the env are different.

```
 if(!isNaN(savedSettings.version) && (savedSettings.version == parseInt(process.env.REACT_APP_SETTINGS_VERSION ?? '1', 10))) {
          dispatch(setSettings(savedSettings));
        }
        else {
          console.warn("Invalid settings configuration, or settings version changed.");
          dispatch(setSettings({...initialSettingsState, ...savedSettings}));
```

File: src/100-App/redux-store.tsx [hunk 2]
```diff
@@ -137,40 +153,85 @@ export const logoutAccount = async(dispatch: (arg0: {type: 'account/resetAccount
...
+};
+
+
+export const saveSettingsMiddleware:Middleware = store => next => action => {
+  const result = next(action);
+
+  if(Object.values(settingsSlice.actions).map(action => action.type).includes(action.type) && action.type !== resetSettings.type) {
+    const settingsState: RootState['settings'] = store.getState().settings;
+    localStorage.setItem('settings', JSON.stringify(settingsState));
+
```
Comment (bohnerjosh) [inline src/100-App/redux-store.tsx:216]: The above code is very clean, nice way to dynamically save settings

File: src/100-App/redux-store.tsx (hunk 1 above)
Comment (ethanjohnsrud) [reply src/100-App/redux-store.tsx]: My thought here was it's safe to add settings.  This way if we add a settings field, it'll get populated with the default and saved on the next iteration; while preserving existing.  When we up the version number we'll be resetting everything.

File: src/100-App/redux-store.tsx (hunk 2 above)
Comment (ethanjohnsrud) [reply src/100-App/redux-store.tsx:216]: Thanks, middlewear gets called on every dispatch for all reducers.  I was looking for a dynamic way to filter; this should work good.

**Review by bohnerjosh — CHANGES_REQUESTED** [review]
Found two issues as discussed on Discord:

1. Unique account route should not query server for non-dirty fields (perform a regex match first locally)
2. "Edit Profile" modal does not appear under account settings

File: src/100-App/redux-store.tsx (hunk 1 above)
Comment (ethanjohnsrud) [reply src/100-App/redux-store.tsx]: Resolved;  good point

**Review by bohnerjosh — APPROVED** [review]
Nice work ethan. Lots of good refactors here

**Comment by ethanjohnsrud** [comment]
[#125 | Portal | Field Updates](https://trello.com/c/LN4JZHYn/125-125-portal-field-updates)

**Pull request merged**